src/test/scala/sigmastate/utxo/ErgoLikeInterpreterSpecification.scala

```diff
@@ -588,4 +588,53 @@ class ErgoLikeInterpreterSpecification extends SigmaTestingCommons {
 
     //todo: check failing branches
   }
+
+  property("PK - P2PK address type") {
+    implicit val ergoAddressEncoder: ErgoAddressEncoder = new ErgoAddressEncoder(ErgoAddressEncoder.TestnetNetworkPrefix)
+
+    val prover = new ErgoLikeProvingInterpreter
+    val verifier = new ErgoLikeInterpreter(networkPrefix = ErgoAddressEncoder.TestnetNetworkPrefix)
+
+    val dk1 = ProveDlog(prover.dlogSecrets.head.publicImage.h)
+    val p2pk = P2PKAddress(dk1)
+    val encodedP2PK = p2pk.toString
+
+    val prop1 = ErgoAddressToSigmaProp(StringConstant(encodedP2PK)).isValid
+
+    val ctx = ErgoLikeContext(
+      currentHeight = 50,
+      lastBlockUtxoRoot = AvlTreeData.dummy,
+      boxesToSpend = IndexedSeq(),
+      ErgoLikeTransaction(IndexedSeq(), IndexedSeq()),
+      self = ErgoBox(20, TrueLeaf, Seq(), Map()))
+
+    val proof1 = prover.prove(prop1, ctx, fakeMessage).get.proof
+    verifier.verify(prop1, ctx, proof1, fakeMessage).map(_._1).getOrElse(false) shouldBe true
+
+    val verifierWrongNetwork = new ErgoLikeInterpreter(networkPrefix = ErgoAddressEncoder.MainnetNetworkPrefix)
+    verifierWrongNetwork.verify(prop1, ctx, proof1, fakeMessage).map(_._1).getOrElse(false) shouldBe false
+  }
+
+  property("PK - P2S address type") {
+    implicit val ergoAddressEncoder: ErgoAddressEncoder = new ErgoAddressEncoder(ErgoAddressEncoder.TestnetNetworkPrefix)
+
+    val prover = new ErgoLikeProvingInterpreter
+    val verifier = new ErgoLikeInterpreter(networkPrefix = ErgoAddressEncoder.TestnetNetworkPrefix)
+
+    val p2s = Pay2SAddress(Values.TrueLeaf)
+    val prop1 = ErgoAddressToSigmaProp(StringConstant(p2s.toString)).isValid
+
+    val ctx = ErgoLikeContext(
+      currentHeight = 50,
+      lastBlockUtxoRoot = AvlTreeData.dummy,
+      boxesToSpend = IndexedSeq(),
+      ErgoLikeTransaction(IndexedSeq(), IndexedSeq()),
+      self = ErgoBox(20, TrueLeaf, Seq(), Map()))
+
+    val proof1 = prover.prove(prop1, ctx, fakeMessage).get.proof
```

Comment:
@ergomorphic Should I spent time on P2S and P2SH address types now? I kind of stuck with `sigmastate.utxo.SigmaPropIsValid cannot be cast to sigmastate.Values$SigmaBoolean` error here. I wonder if it worth spending much time or should I skip them and get back to it later.

Reply:
@ergomorphic nevermind, I scraped that off. I saw your comment about missing sigma prop types for P2S and P2SH.